### ISSUE TITLE
Build issues on Debian 9

### ISSUE THREAD

**Issue by zwoop** [apache/trafficserver#2167]
Seeing:

```
SSLUtils.cc: In function 'void SSLInitializeLibrary()':
SSLUtils.cc:869:57: error: statement has no effect [-Werror=unused-value]
     CRYPTO_THREADID_set_callback(SSL_pthreads_thread_id);
                                                         ^
SSLUtils.cc: At global scope:
SSLUtils.cc:151:1: error: 'void SSL_locking_callback(int, int, const char*, int)' defined but not used [-Werror=unused-function]
 SSL_locking_callback(int mode, int type, const char *file, int line)
 ^~~~~~~~~~~~~~~~~~~~
SSLUtils.cc:143:1: error: 'void SSL_pthreads_thread_id(CRYPTO_THREADID*)' defined but not used [-Werror=unused-function]
 SSL_pthreads_thread_id(CRYPTO_THREADID *id)
 ^~~~~~~~~~~~~~~~~~~~~~
```

**Comment by zwoop** [comment]
Looks like CRYPTO_THREADID_set_callback() is a no-op ?

**Comment by shinrich** [comment]
As I recall, they eliminated the need for the thread ids in 1.1, and they reworked the locking considerably.

**Comment by SolidWallOfCode** [comment]
Fixed by #2255.
